## 1. Introdução

O presente exercício consiste na análise de uma linha de transmissão trifásica em duas configurações distintas: linha sem transposição e linha com transposição.

## 2. Dados da rede

A Figura 1 apresenta a matriz de impedâncias série da linha sem transposição. Note que somente o triângulo superior da matriz é indicado, já que a matriz é simétrica.



![](https://github.com/Zolubas/Eletric-Power-Systems/blob/master/matrizdeimpedanciasseriedalinhasemtransposicaoemohmporkm.png?raw=true.png)

### Dados da rede
* Comprimento da linha = 200 km
* Corrente absorvida pela carga equilibrada, alimentada pela linha. Em [A], em cada fase = 200A

### 3. Questões

* 1) Calcular o módulo da queda de tensão por fase na linha sem transposição, em [V] (3 valores).

* 2) Calcular a perda ativa total (3 fases) na linha sem transposição, em [MW].

* 3) Calcular o módulo da queda de tensão por fase na linha completamente transposta, em [V] (3 valores).

* 4) Calcular a perda ativa total (3 fases) na linha completamente transposta, em [MW].

In [55]:
import cmath
import numpy as np
K = 1000
M = K*K


In [56]:
Z = 200*np.array([[0.0846 + 0.6423j , 0.0567 + 0.3308j , 0.0573 + 0.3499j],
    [0.0567 + 0.3308j , 0.0833 + 0.6438j , 0.0567 + 0.3308j],
    [0.0573 + 0.3499j ,  0.0567 + 0.3308j , 0.0846 + 0.6423j]])


In [57]:
def phase(x): # em graus
    return ((180/np.pi)*(np.arctan2(x.imag,x.real)))

def modulo(x):
    return np.sqrt(x.real*x.real + x.imag*x.imag)
            
def polar(x):
    print(str(modulo(x)) + "/_" + str(phase(x)))
    
def cart2pol(x, y):
    rho = np.sqrt(x**2 + y**2)
    phi = 180*np.arctan2(y, x)/np.pi
    return(rho, phi)

def pol2cart(rho, phi):
    x = rho * np.cos(phi)
    y = rho * np.sin(phi)
    return x + y*1j 


In [58]:
alpha = -0.4999999999999998 + 0.8660254037844387j

In [59]:
I = 200*np.array([1 , alpha**2, alpha])
polar(I)

[200. 200. 200.]/_[   0. -120.  120.]


In [60]:
polar(alpha)

0.9999999999999999/_119.99999999999999


# 1) Calculo do modulo da queda de tensão na linha por fase (sem transposição de fases) 

In [61]:
V = Z@I

In [62]:
V

array([   442.35659151+12098.78460969j,  10310.63805538 -7181.45102963j,
       -11324.67653115 -4499.51564938j])

In [63]:
V.shape

(3,)

In [64]:
polar(V)

[12106.86864494 12565.13016248 12185.80894378]/_[  87.90608049  -34.85755422 -158.33110683]


#### Questão 1
* VAA' = 12106.86864494 /_ 87.90608049
* VBB' = 12565.13016248 /_ -34.85755422
* VCC' = 12185.80894378 /_ -158.33110683

In [65]:
modulo(V)

array([12106.86864494, 12565.13016248, 12185.80894378])

In [66]:
def diferenca_de_phase(a,b):
    return abs(b - a)
def dif(a,b):
    print(diferenca_de_phase(phase(V[a]),phase(V[b])))

In [88]:
print(-34.85755422+34.85755422)
print(-158.33110683+34.85755422)
print(87.90608049+34.85755422)

0.0
-123.47355261000001
122.76363470999999


Vemos que as diferenças de fase que deveriam ser exatamente 120 graus
não bate com a realidade(-123 e 122). Sendo assim propõe-se a  transposição de fase
a ser estudada nos itens 3) e 4)



In [93]:
#Erro:
EBBlinha = 100*(123.47355261000001-120)/120 
ECClinha = 100*(122.76363470999999-120)/120
print("Erro na linha B = "+ str(EBBlinha) + " %")
print("Erro na linha C = "+ str(ECClinha) + "%")

Erro na linha B = 2.8946271750000108 %
Erro na linha C = 2.3030289249999925%


# 2) Calculo da potência (sem transposição de fases)

In [68]:
def FP(phaseV,phaseI):
    return np.cos(np.pi*(phaseV - phaseI)/180)
P = (V*I.conjugate()).real/M
print(P) #potencia ativa em MW

[0.08847132 0.2128     0.35312868]


In [96]:
print("Perda Ativa total = " + str(P[0] + P[1] + P[2]) +" " + "MW")

Perda Ativa total = 0.6543999999999994 MW


Assim vemos que
* PAA' = 0.08847132 MW
* PBB' = 0.2128 MW
* PCC' = 0.35312868

quando na verdade deveriamos ter P0 = P1 = P2

# 3) Calculando o modulo da queda de tensão na linha por fase (com transposição de fase)


 ![](https://github.com/Zolubas/Eletric-Power-Systems/blob/master/TransposicaoDefase1.png?raw=true.png)

Vemos que nesse caso cada mutua na matriz Z afetaria o sistema multiplicada por um fator 1/3. Alem disso todas fases
AA', BB', CC' experimentariam as mesmas Mutuas, o que tenderia a equilibrar mais o sistema com relacção a parte imaginaria
dos termos que não se encontram na diagonal principal da matriz Z. Porém vejamos que efeito isso tem na parte real desses
valores haja vista que eles modelam não as mutuas mas sim efeitos de perda por retorno pela terra que é composta por
diferentes densidades e extratos nos 200 km de linha considerados.

In [70]:
#Mutua média ZM , A[LINHA][COLUNA]
# ZM = (MAB + MAC + MBC)/3
ZM = 200*((0.0567 + 0.3308j) + (0.0573 + 0.3499j) + (0.0567 + 0.3308j))/3

In [71]:
ZM

(11.38+67.43333333333332j)

In [72]:
ZA = (0.0846 + 0.6423j)*200
ZB = (0.0833 + 0.6438j)*200
ZC = (0.0846 + 0.6423j)*200
ZP = (ZA + ZB + ZC)/3
Z_com_transposicao = np.array([[ZP,ZM,ZM],[ZM,ZP,ZM],[ZM,ZM,ZP]])

In [79]:
ZP/200

(0.08416666666666667+0.6428000000000001j)

In [80]:
Z_com_transposicao

array([[16.83333333+128.56j      , 11.38       +67.43333333j,
        11.38       +67.43333333j],
       [11.38       +67.43333333j, 16.83333333+128.56j      ,
        11.38       +67.43333333j],
       [11.38       +67.43333333j, 11.38       +67.43333333j,
        16.83333333+128.56j      ]])

In [81]:
V_com_transposicao = Z_com_transposicao@I
polar(V_com_transposicao)

[12273.88809175 12273.88809175 12273.88809175]/_[  84.90193083  -35.09806917 -155.09806917]


In [82]:
modulo(V_com_transposicao)

array([12273.88809175, 12273.88809175, 12273.88809175])

In [87]:
print(-35.09806917 +35.09806917 )
print(-155.09806917+35.09806917)
print(84.90193083+35.09806917)

0.0
-120.0
120.0


In [94]:
#Erro:
EBBlinha = 100*(120.0-120)/120 
ECClinha = 100*(120.0-120)/120
print("Erro na linha B = "+ str(EBBlinha) + " %")
print("Erro na linha C = "+ str(ECClinha) + "%")

Erro na linha B = 0.0 %
Erro na linha C = 0.0%


# 4) Potência Ativa total com transposição

In [95]:
P = (V_com_transposicao*I.conjugate()).real/M
print(P) #potencia ativa em MW

[0.21813333 0.21813333 0.21813333]


In [97]:
print("Perda Ativa total = " + str(P[0] + P[1] + P[2]) +" " + "MW")

Perda Ativa total = 0.6543999999999994 MW
